packages/kbn-i18n/src/core/i18n.test.js

```diff
@@ -0,0 +1,737 @@
+    test('should return "en" locale by default', () => {
+      expect(i18n.getDefaultLocale()).toBe('en');
+    });
+
+    test('should return updated locale', () => {
+      i18n.setDefaultLocale('foo');
+      expect(i18n.getDefaultLocale()).toBe('foo');
+    });
+  });
+
+  describe('setFormats', () => {
+    test('should throw error if formats parameter is not an object', () => {
+      expect(() => i18n.setFormats(undefined)).toThrow();
+      expect(() => i18n.setFormats(null)).toThrow();
+      expect(() => i18n.setFormats(true)).toThrow();
+      expect(() => i18n.setFormats(5)).toThrow();
+      expect(() => i18n.setFormats('foo')).toThrow();
+    });
+
+    test('should throw error if formats parameter is empty object', () => {
+      expect(() => i18n.setFormats({})).toThrow();
+    });
+
+    test('should merge current formats with a passed formats', () => {
+      expect(i18n.getFormats().date.short).not.toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'short',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'short',
+        day: 'numeric',
+        year: 'numeric',
+      });
+
+      i18n.setFormats({
+        date: {
+          short: {
+            month: 'long',
+            day: 'numeric',
+            year: 'numeric',
+          },
+        },
+      });
+
+      expect(i18n.getFormats().date.short).toEqual({
+        month: 'long',
+        day: 'numeric',
+        year: 'numeric',
+      });
+    });
+  });
+
+  describe('getFormats', () => {
+    test('should return "en" formats by default', () => {
+      const { formats } = require('./formats');
+
+      expect(i18n.getFormats()).toEqual(formats);
+    });
+
+    test('should return updated formats', () => {
+      const { formats } = require('./formats');
+
+      i18n.setFormats({
+        foo: 'bar',
+      });
+
+      expect(i18n.getFormats()).toEqual({
+        ...formats,
+        foo: 'bar',
+      });
+    });
+  });
+
+  describe('getRegisteredLocales', () => {
+    test('should return empty array by default', () => {
+      expect(i18n.getRegisteredLocales()).toEqual([]);
+    });
+
+    test('should return array of registered locales', () => {
+      i18n.addMessages({
+        locale: 'en',
+      });
+
+      expect(i18n.getRegisteredLocales()).toEqual(['en']);
+
+      i18n.addMessages({
+        locale: 'ru',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru');
+      expect(i18n.getRegisteredLocales().length).toBe(2);
+
+      i18n.addMessages({
+        locale: 'fr',
+      });
+
+      expect(i18n.getRegisteredLocales()).toContain('en', 'ru', 'fr');
+      expect(i18n.getRegisteredLocales().length).toBe(3);
+    });
+  });
+
+  describe('translate', () => {
```

Comment:
__issue:__ I believe we need a couple of tests for locales with country code to see how we fallback to "generic" locale, let's say we have three labels with the following ids: `one`, `two` and `three` and registered two localization files with the following content (pseudo format):
```
locale: ru
two: 'два'
three: 'три'
```
```
locale: ru-BY:
three: 'тры'
```
For `one` default message is `one`, so if we set locale to `ru-BY` I guess we should get the following:
```
i18n.setLocale('ru-BY');
i18n.translate('one') => 'one'
i18n.translate('two') => 'два'
i18n.translate('three') => 'тры'
```
What do you think? Do we support that scenario at all (I think we were discussing that, but it somehow slipped under my radar)? If not, it's not a blocker we can discuss it later, but test is still needed.

Reply:
added skipped test with `TODO` action for now, but let's maybe discuss it offline